From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nopjmp <kthompson@hey.com>
Date: Wed, 10 Feb 2021 14:37:35 -0600
Subject: [PATCH] Fix Dispenser Shulker Crash

This will now check to make sure the shulker box was actually placed before interacting with it.

diff --git a/src/main/java/net/minecraft/server/DispenserRegistry.java b/src/main/java/net/minecraft/server/DispenserRegistry.java
index fb1d640cc3e133c7623d051893fd85324b2c4de4..c3d2d7947b33b2331cfd61a325277eebe1240770 100644
--- a/src/main/java/net/minecraft/server/DispenserRegistry.java
+++ b/src/main/java/net/minecraft/server/DispenserRegistry.java
@@ -757,7 +757,12 @@ public class DispenserRegistry {
                 EnumDirection enumdirection1 = world.isEmpty(blockposition.down()) ? enumdirection : EnumDirection.UP;
                 IBlockData iblockdata = block.getBlockData().set(BlockShulkerBox.a, enumdirection1);
 
-                world.setTypeUpdate(blockposition, iblockdata);
+                // Dionysus start - fix Dispenser crashes
+                boolean wasPlaced = world.setTypeUpdate(blockposition, iblockdata);
+                if (!wasPlaced) {
+                    return itemstack;
+                }
+                // Dionysus end
                 TileEntity tileentity = world.getTileEntity(blockposition);
                 ItemStack itemstack1 = itemstack.cloneAndSubtract(1);
 
